### DISCUSSION TITLE
Changing docs system

### DISCUSSION THREAD

**Discussion by dolby360** [TomerFi/aioswitcher#510] [Q&A] [answered]
Hi.
I'm working on [this issue](https://github.com/TomerFi/aioswitcher/issues/442). And I'm trying to figure out what is the equivalent of  `.. autoprogram:: discover_devices:parser` [like here](https://github.com/TomerFi/aioswitcher/blob/dev/docs/scripts.rst). In mkdocs.
In other words, how do I display argparse nicely in mkdocs.
[This is what I did so far](https://github.com/dolby360/aioswitcher/tree/dolev/better_docs).

@TomerFi

**Comment by TomerFi** [comment, marked as answer]
Actually, I was thinking we should use [mkdocs-material](https://squidfunk.github.io/mkdocs-material/), it has a ton of plugins/extensions and I'm sure it has a one for auto-generating docs from code as well, although I'm not sure we'll need it.

Please take a look on 2 other projects of mine where I used [mkdocs-material](https://squidfunk.github.io/mkdocs-material/):
site: https://archetypes.tomfi.info/ - repo: https://github.com/TomerFi/tomfi-archetypes
site: https://auto-me-bot.tomfi.info/ - repo: https://github.com/TomerFi/auto-me-bot

**Reply by dolby360** [reply]
Exactly what I did. I just can't find how to represent the arguments (in argparse). I'm still searching, but if you know it will help a lot.

**Reply by TomerFi** [reply]
I'll look around, perhaps I'll bump into a relevant extension.
But don't feel obligated to go that path, API functions in this module rarely change,
so automating this might be redundant, feel free to use code blocks, etc.
